src/snowflake/connector/config_parser/__init__.py

```diff
@@ -0,0 +1,32 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
+
+from pathlib import Path
+from shutil import copyfile
+
+from tomlkit import parse
+
+from ..constants import config_file
+from .parser import ConfigParser
+
+if not config_file.exists():
+    # Create default config file
+    default_config = Path(__file__).absolute().parent / "default_config.toml"
+    copyfile(default_config, config_file)
+
+CONFIG_PARSER = ConfigParser(
+    name="CONFIG_PARSER",
+    file_path=config_file,
+)
+CONFIG_PARSER.add_option(
+    name="connections",
+    _type=parse,
+)
```

Comment:
What do you think about encapsulating this in factory method like `_build_config_parser`?

Reply:
I think we don't need to do that yet, I'm a fan of only complicating the code when there's a need.
Something like what you're suggesting is usually useful for testing.

I'm not against it though. If others feel like we should do this from the get-go we can!

Reply:
I've now done this because I needed it for testing